src/modpack/files.nim

```diff
@@ -36,22 +39,32 @@ const
   manifestFile* = joinPath(packFolder, "manifest.json")
   outputFolder* = joinPath(projectFolder, ".out/")
 
-proc initManifestFile(projectId: int, fileId: int): ManifestFile =
+proc initManifestFile*(projectId: int, fileId: int, name: string,
+    explicit: bool, dependencies: seq[int]): ManifestFile =
   ## create a new manifest fmod object.
   result.projectId = projectId
   result.fileId = fileId
+  result.name = name
+  result.explicit = explicit
+  result.dependencies = dependencies
 
 converter toManifestFile(json: JsonNode): ManifestFile =
   ## creates a ManifestFile from manifest json
   result.projectId = json["projectID"].getInt()
   result.fileId = json["fileID"].getInt()
+  result.name = json["name"].getStr()
+  result.explicit = json["explicit"].getBool()
+  result.dependencies = json["dependencies"].getElems().map((x) => x.getInt())
 
 converter toJson(file: ManifestFile): JsonNode {.used.} =
   ## creates the json for a manifest file `file`
   result = %* {
     "projectID": file.projectId,
     "fileID": file.fileId,
-    "required": true
+    "required": true,
+    "name": file.name,
+    "explicit": file.explicit,
+    "dependencies": file.dependencies
```

Comment:
I'd put the pax-specific properties in a `__paxMeta` or `__paxMetadata` map like discussed.
It's a bit more lines in the manifest, but it feels cleaner to me and we can more easily add a feature to `./pax export` that would purge those.